### ISSUE TITLE
Prefix all components to use <CXxxx /> or <c-xxxx /> pattern

### ISSUE THREAD

**Issue by codebender828** [chakra-ui/chakra-ui-vue#83]
**Is your feature request related to a problem? Please describe.**
In conformance to the Vue style guide for components, component names/tags should always be Multi-Word. This helps prevent name clashing with HTML elements since Vue uses templates very similar to HTML elements.
> Component names should always be multi-word, except for root App components, and built-in components provided by Vue, such as <transition> or <component>
> ~ [Vue style guide](https://vuejs.org/v2/style-guide/#Multi-word-component-names-essential)

**Describe the solution you'd like**
This involves prefixing all components with the `c-` or `C`
e.g.
`Accordion` -> `CAccordion` or `c-accordion`
`Button` -> `CButton` or `c-button`

**Comment by koca** [comment]
I think we can just change the exports in `src/index` file. 
https://github.com/chakra-ui/chakra-ui-vue/blob/86592740ec5e6c7826d8d9fa48375dd6ec8b9d7b/packages/chakra-ui-core/src/index.js#L20

```diff 
+CBox
-Box
```
If we do this they can use both `c-box` and `CBox`.

**Comment by koca** [comment]
And It sucks i like components without prefix. But vue :(

**Comment by codebender828** [comment]
Yeah, we could serve the imports like that, but the problem is that Vue's component resolver will still complain if it finds the name attribute of each component with something non-prefixed on a few elements. Like the Image or the Switch components. I tried this and it was still complaining. So I just decided to rename everything and cleanup a bit. Will be pushing in a few.

So it's just better to do it for all of them.
